.github/workflows/test.yaml

```
@@ -0,0 +1,20 @@
+name: Go test
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+jobs:
+  test:
+    name: Go Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.19"
```

[nit] we could use the minimum version to install. We don't need to pin a specific version for this project. 

```suggestion
      - uses: actions/setup-go@v3
        with:
          go-version: '>=1.19.0'
```